vaadin-login-flow-parent/vaadin-login-flow/src/main/java/com/vaadin/flow/component/login/LoginOverlay.java

```diff
@@ -122,10 +127,22 @@ private UI getCurrentUI() {
     private void ensureAttached() {
         if (getElement().getNode().getParent() == null) {
             UI ui = getCurrentUI();
-            ui.beforeClientResponse(ui, context -> {
-                ui.addToModalComponent(this);
-                autoAddedToTheUi = true;
-            });
+            StateTree.ExecutionRegistration addToUiRegistration = ui
+                    .beforeClientResponse(ui, context -> {
+                        ui.addToModalComponent(this);
+                        autoAddedToTheUi = true;
```

Comment:
Should we add `afterProgrammaticNavigationListenerRegistration.remove()` here also? Otherwise, the listener will stay there, invoking on every navigation event, until a programmatic navigation occurs, right?

Reply:
Definitely. Updated the PR.